taqueria-plugin-contract-types/tasks.ts

```diff
@@ -0,0 +1,128 @@
+    //             ? parsedArgs.config.environment.default
+    //             : 'development'
+    //     )
+    // const env = parsedArgs.config.environment && parsedArgs.config.environment[currentEnv]
+    //         ? parsedArgs.config.environment[currentEnv]
+    //         : undefined
+            
+    // // Has storage been provided for this contract?
+    // if (env && env.storage) {
+    //     try {
+    //         const tezos = new TezosToolkit(env.rpcUrl)
+    //         const contractData = await readFile(contractAbspath, "utf-8")
+
+    //         // TODO: Generate contract michelson
+    //         // TODO: Generate types from michelson
+
+    //         throw new Error('Not Implemented');
+    //         // await importKey(tezos, env.faucet.email, env.faucet.password, env.faucet.mnemonic.join(' '), env.faucet.activation_code)
+    //         // return tezos.contract.originate({
+    //         //     code: contractData,
+    //         //     storage: env.storage[contractFilename]
+    //         // })
+    //         // .then(operation => `${contractFilename}: ${operation.contractAddress}`)
+    //     }
+    //     catch (err) {
+    //         return Promise.reject({
+    //             status: 'failed',
+    //             stdout: "",
+    //             stderr: err
+    //         })
+    //     }
+    // }
+
+    // return Promise.reject({
+    //     status: 'failed',
+    //     stderr: `No storage configured in your configuration file for ${contractFilename}`,
+    //     stdout: ""
+    // })
+}
+
+const generateContractTypesAll = (parsedArgs: Opts & PluginOpts) : Promise<string[]> =>
+    glob("**/*.tz", {cwd: parsedArgs.artifactsDir})
+    .then(files => Promise.all(files.map(generateContractTypes(parsedArgs))))
+
+export const generateTypes = <T>(parsedArgs: Opts): LikeAPromise<ActionResponse, Failure<T>> => {
+    parsedArgs.typescriptDir = parsedArgs.typescriptDir || 'types';
+    // if(!parsedArgs.typescriptDir){
+    //     return Promise.reject({
+    //         status: 'failed',
+    //         stderr: `No typescriptDir configured`,
+    //         stdout: ""
+    //     });
+    // }
+
+    // WORKAROUND: Redirect console.log
+    const strOutLog = [] as string[];
+    const consoleLogOrig = console.log;
+    console.log = (message:string, data?:unknown) => {
+        strOutLog.push(`${message}${data?`\n${JSON.stringify(data,null,2)}`:''}`);
+    }
+    console.log('generateTypes', { 
+        typescriptDir: parsedArgs.typescriptDir
+    });
+
+    // console.log = consoleLogOrig;
+    // return Promise.resolve({
+    //     status: 'success',
+    //     stdout: `${strOutLog.join('\n')}`,
+    //     stderr: ""
+    // });
+
+    const argsTyped = parsedArgs as Opts & PluginOpts;
+
+    const p = argsTyped.contract
+        ? generateContractTypes(argsTyped) (argsTyped.contract as string)
+        : generateContractTypesAll(argsTyped)
+
+    return p.then(data => {
```

Comment:
Taqueria has the ability to render objects in a table format. Should that be used here for UI consistency?

Reply:
I'll create a separate task for that, it will require a bit of work to go through all the console.log to clean everything up.

Reply:
https://github.com/ecadlabs/taqueria/issues/332